### PR TITLE
Enable default full disk encryption(FDE)

### PR DESCRIPTION
**Pull request by tingai1**
1.Add FDE config flag, and point default disk encryption via metadata partition
2.Update init to support FDE flow
3.Open some kernel config used by FDE feauture

Jira: None.
Test: 1.Data partition can be encryption and system can boot to home screen.

Signed-off-by: Ai, Ting A <ting.a.ai@intel.com>

### CODE REVIEW

File: androidia_64/fstab [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 /dev/block/mmcblk1p7	/system         ext4    ro															wait
 /dev/block/mmcblk1p8	/cache          ext4    noatime,nosuid,nodev,errors=panic                           wait,check
-/dev/block/mmcblk1p12	/data           ext4    noatime,nosuid,nodev,discard,noauto_da_alloc,errors=panic   wait,check,formattable
+/dev/block/mmcblk1p12	/data           ext4    noatime,nosuid,nodev,discard,noauto_da_alloc,errors=panic   wait,check,formattable,forceencrypt=/dev/block/mmcblk1p6
```
Comment (kalyankondapally) [inline androidia_64/fstab]: should this be mmcblk1p12 ?

**Comment by kalyankondapally** [comment]
@Kishore409 for checking it on joule

**Comment by kalyankondapally** [comment]
@Kishore409 This is not ready for checking yet, pls ignore till I update

File: androidia_64/fstab (hunk 1 above)
Comment (fofeir) [inline androidia_64/fstab]: Hi Kalyan,
mmcblk1p6 maps to the /meta which is the right partition for encryption. This is NOT a typo.  :)

**Comment by tpalli** [comment]
I have a newbie question: What's the actual effect of this change? Can I still freely push/pull stuff to the disk via adb etc? What exactly is being encrypted?

**Comment by tingai1** [comment]
@tpalli FDE actually encrypt /data partition, and it mainly avoid chip-off attack, and if you can login to desktop, that means /data has been decrypted, so you can push/pull via adb if you have permission.

**Comment by kalyankondapally** [comment]
@Kishore409 Can u check this for Joule

**Comment by Kishore409** [comment]
@kalyankondapally as per the test suggested by @tingai1 the patch is enabling disk encryption
androidia_64:/ # getprop | grep rypt
[init.svc.uncrypt]: [stopped]
[ro.crypto.fs_crypto_blkdev]: [/dev/block/dm-0]
[ro.crypto.state]: [encrypted]
[ro.crypto.type]: [block]
[vold.decrypt]: [trigger_restart_framework]
[vold.encrypt_progress]: [100]
[vold.encrypt_time_remaining]: [0]
androidia_64:/ #

**Comment by kalyankondapally** [comment]
can install mode with Grub be tested also, seems to break my SKL. Cannot go past animation screen

**Comment by kalyankondapally** [comment]
Ya first boot took some time but subsequent boots are fine

**Comment by kalyankondapally** [comment]
Ya first boot took some time but subsequent boots are fine

**Comment by Kishore409** [comment]
@tingai1 @fofeir Merged, Thanks

**Pull request merged**